index.js

```diff
@@ -149,6 +151,16 @@ const Socket = exports.Socket = class TCPSocket extends Duplex {
     return this
   }
 
+  setTimeout (ms, ontimeout) {
+    if (ontimeout) this.on('timeout', ontimeout)
```

Comment:
```suggestion
    if (ontimeout) this.once('timeout', ontimeout)
```

So it'll only trigger when _this_ timeout is reached, not necessarily any future timeouts.